cocos/scripting/lua-bindings/manual/network/lua_downloader.cpp

```diff
@@ -335,13 +337,16 @@ static const struct luaL_Reg downloaderMemberFns[] = {
 
 int register_downloader(lua_State* L)
 {
-    luaL_newmetatable(L, "cc.Downloader");
-    lua_pushstring(L, "__index");
-    lua_pushvalue(L, -2);    /* pushes the metatable */
-    lua_settable(L, -3);     /* metatable.__index = metatable */
-
-    luaL_openlib(L, nullptr, downloaderMemberFns, 0);
-    luaL_openlib(L, "cc.Downloader", downloaderStaticFns, 0);
+    int stackSize = lua_gettop(L);
+    luaL_newmetatable(L, "cc.Downloader");  //stack metatable(cc.Downloader)
+    lua_pushstring(L, "__index");           //stack metatable(*), __index
+    lua_pushvalue(L, -2);                   //stack metatable(*), __index, metatable(*)
+    lua_settable(L, -3);                    //stack metatable(*)
+    luaL_register(L, nullptr, downloaderMemberFns);   //stack metatable(*)
+    lua_pop(L, 1);                                      //stack *empty*
+    luaL_register(L, "cc.Downloader", downloaderStaticFns);   //stack *empty*
+    lua_pop(L, 1);
```

Comment:
stack clean up may fix this issue https://github.com/cocos2d/cocos2d-x/issues/19253